src/engine/TransitivePathBinSearch.cpp

```diff
@@ -0,0 +1,245 @@
+    size_t minDist, size_t maxDist)
+    : TransitivePathBase(qec, child, leftSide, rightSide, minDist, maxDist) {
+  auto [startSide, targetSide] = decideDirection();
+  subtree_ = QueryExecutionTree::createSortedTree(
+      subtree_, {startSide.subCol_, targetSide.subCol_});
+}
+
+// _____________________________________________________________________________
+template <size_t RES_WIDTH, size_t SUB_WIDTH, size_t SIDE_WIDTH>
+void TransitivePathBinSearch::computeTransitivePathBound(
+    IdTable* dynRes, const IdTable& dynSub, const TransitivePathSide& startSide,
+    const TransitivePathSide& targetSide, const IdTable& startSideTable) const {
+  auto timer = ad_utility::Timer(ad_utility::Timer::Stopped);
+  timer.start();
+
+  auto [edges, nodes] = setupMapAndNodes<SUB_WIDTH, SIDE_WIDTH>(
+      dynSub, startSide, targetSide, startSideTable);
+
+  timer.stop();
+  auto initTime = timer.msecs();
+  timer.start();
+
+  Map hull(allocator());
+  if (!targetSide.isVariable()) {
+    hull = transitiveHull(edges, nodes, std::get<Id>(targetSide.value_));
+  } else {
+    hull = transitiveHull(edges, nodes, std::nullopt);
+  }
+
+  timer.stop();
+  auto hullTime = timer.msecs();
+  timer.start();
+
+  fillTableWithHull(*dynRes, hull, nodes, startSide.outputCol_,
+                    targetSide.outputCol_, startSideTable,
+                    startSide.treeAndCol_.value().second);
+
+  timer.stop();
+  auto fillTime = timer.msecs();
+
+  auto& info = runtimeInfo();
+  info.addDetail("Initialization time", initTime.count());
+  info.addDetail("Hull time", hullTime.count());
+  info.addDetail("IdTable fill time", fillTime.count());
+}
+
+// _____________________________________________________________________________
+template <size_t RES_WIDTH, size_t SUB_WIDTH>
+void TransitivePathBinSearch::computeTransitivePath(
+    IdTable* dynRes, const IdTable& dynSub, const TransitivePathSide& startSide,
+    const TransitivePathSide& targetSide) const {
+  auto timer = ad_utility::Timer(ad_utility::Timer::Stopped);
+  timer.start();
+
+  auto [edges, nodes] =
+      setupMapAndNodes<SUB_WIDTH>(dynSub, startSide, targetSide);
+
+  timer.stop();
+  auto initTime = timer.msecs();
+  timer.start();
+
+  Map hull{allocator()};
+  if (!targetSide.isVariable()) {
+    hull = transitiveHull(edges, nodes, std::get<Id>(targetSide.value_));
+  } else {
+    hull = transitiveHull(edges, nodes, std::nullopt);
+  }
+
+  timer.stop();
+  auto hullTime = timer.msecs();
+  timer.start();
+
+  fillTableWithHull(*dynRes, hull, startSide.outputCol_, targetSide.outputCol_);
+  timer.stop();
+  auto fillTime = timer.msecs();
+
+  auto& info = runtimeInfo();
+  info.addDetail("Initialization time", initTime.count());
+  info.addDetail("Hull time", hullTime.count());
+  info.addDetail("IdTable fill time", fillTime.count());
+}
+
+// _____________________________________________________________________________
+ResultTable TransitivePathBinSearch::computeResult() {
```

Comment:
This also is probably almost the same for both implementations.
So you basically just need to template on the `Map` type, and then you can get rid of all the duplications.
Let me know if there's questions.